test/metric/dimension/instanceid_provider.go

```diff
@@ -0,0 +1,67 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package dimension
+
+import (
+	"github.com/aws/amazon-cloudwatch-agent-test/environment/computetype"
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/awsservice"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+	"log"
+)
+
+type ECSInstanceIdDimensionProvider struct {
+	Provider
+}
+
+var _ IProvider = (*ECSInstanceIdDimensionProvider)(nil)
+
+func (p *ECSInstanceIdDimensionProvider) IsApplicable() bool {
+	if p.env.ComputeType == computetype.ECS {
+		return true
+	}
+	return false
+}
+
+func (p *ECSInstanceIdDimensionProvider) GetDimension(instruction Instruction) types.Dimension {
+	if instruction.Key != "InstanceId" || instruction.Value.IsKnown() {
+		return types.Dimension{}
+	}
+
+	//TODO currently assuming there's only one container
+	containerInstances, err := awsservice.GetContainerInstances(p.env.EcsClusterArn)
+	if err != nil {
+		log.Print(err)
+		return types.Dimension{}
+	}
+
+	return types.Dimension{
+		Name:  aws.String("InstanceId"),
+		Value: aws.String(containerInstances[0].EC2InstanceId),
+	}
+}
+
+type InstanceIdDimensionProvider struct {
+	Provider
+}
+
+var _ IProvider = (*ECSInstanceIdDimensionProvider)(nil)
```

Comment:
should this be `InstanceIdDimensionProvider`?

Reply:
crab thank you